core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -219,6 +220,19 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
     }
   }
 
+  override def getEventLogPaths(appId: String, attemptId: Option[String]): Seq[Path] = {
+
+    val filePaths = new ArrayBuffer[Path]()
+    applications.get(appId).foreach { appInfo =>
```

Comment:
I think this is a little more scala-y and avoids the intermediate array:

    applications.get(appId).flatMap { appInfo =>
      // If no attempt is specified, or there is no attemptId for attempts, return all attempts
      appInfo.attempts
        .filter { attempt =>
          attempt.attemptId.isEmpty || attemptId.isEmpty || attempt.attemptId.get == attemptId.get
        }
        .map { attempt => new Path(logDir, attempt.logPath) }
    }

Also, probably not an issue in practice, but `attempt.attemptId.isEmpty` should probably be removed from the filter condition.